library/agent/Agent.ts

```diff
@@ -36,6 +36,7 @@ export class Agent {
   private preventedPrototypePollution = false;
   private incompatiblePackages: Record<string, string> = {};
   private wrappedPackages: Record<string, WrappedPackage> = {};
+  private requiredPackages: Record<string, string[]> = {};
```

Comment:
It doesn't make much difference, but perhaps we should use maps more often in the lib.
https://www.zhenghao.io/posts/object-vs-map